File: backend/app/schemas/paper.py
from pydantic import BaseModel, EmailStr
from typing import List, Optional
from enum import Enum


class TemplateId(str, Enum):
    IEEE_CONFERENCE = "ieee-conference"
    IEEE_JOURNAL = "ieee-journal"
    SPRINGER = "springer"
    ACM = "acm"
    ELSEVIER = "elsevier"


class Author(BaseModel):
    name: str
    affiliation: str
    department: str
    email: str


class PaperSections(BaseModel):
    abstract: str = ""
    introduction: str = ""
    literatureReview: str = ""
    methodology: str = ""
    systemDesign: str = ""
    implementation: str = ""
    results: str = ""
    conclusion: str = ""
    futureScope: str = ""
    acknowledgement: str = ""
    references: str = ""


class PaperCreate(BaseModel):
    title: str
    authors: List[Author]
    keywords: List[str]
    templateId: TemplateId
    sections: PaperSections


class PaperResponse(BaseModel):
    id: str
    title: str
    authors: List[Author]
    keywords: List[str]
    templateId: TemplateId
    sections: PaperSections
    message: str = "Paper saved successfully"


class GenerateRequest(BaseModel):
    paper: PaperCreate
    format: str = "docx"  # "docx" or "pdf"


class GenerateResponse(BaseModel):
    download_url: str
    filename: str
    format: str
    message: str = "Document generated successfully"

File: frontend/src/types/paper.ts
// ─── Template Types ──────────────────────────────────────────

export type TemplateId =
  | "ieee-conference"
  | "ieee-journal"
  | "springer"
  | "acm"
  | "elsevier";

export interface TemplateConfig {
  id: TemplateId;
  name: string;
  description: string;
  thumbnail: string;
  columns: 1 | 2;
  fontFamily: string;
  fontSize: {
    title: string;
    authors: string;
    sectionHeading: string;
    body: string;
    caption: string;
    references: string;
  };
  margins: {
    top: string;
    bottom: string;
    left: string;
    right: string;
  };
  lineSpacing: number;
  sectionNumbering: boolean;
  tags: string[];
}

// ─── Paper Types ─────────────────────────────────────────────

export interface Author {
  name: string;
  affiliation: string;
  department: string;
  email: string;
}

export type SectionKey =
  | "abstract"
  | "introduction"
  | "literatureReview"
  | "methodology"
  | "systemDesign"
  | "implementation"
  | "results"
  | "conclusion"
  | "futureScope"
  | "acknowledgement"
  | "references";

export interface PaperSection {
  key: SectionKey;
  label: string;
  content: string;
  order: number;
}

export interface Paper {
  id?: string;
  title: string;
  authors: Author[];
  keywords: string[];
  templateId: TemplateId;
  sections: Record<SectionKey, string>;
  createdAt?: string;
  updatedAt?: string;
}
